lib/routes/ecnu/cxcy.ts

```diff
@@ -0,0 +1,90 @@
+import { load } from 'cheerio';
+
+import type { Route } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+
+export const route: Route = {
+    path: '/cxcy/:type?',
+    categories: ['university'],
+    example: '/ecnu/cxcy',
+    parameters: { type: '默认为 announcement' },
+    radar: [
+        {
+            source: ['cxcy.ecnu.edu.cn'],
+            target: '/cxcy',
+        },
+    ],
+    name: '本科创新创业教育网',
+    maintainers: ['FrozenStarrrr', 'ChiyoYuki', 'ECNU-minus'],
+    description: `| 通知公告     | 新闻动态     | 学科竞赛     | 常用资源     |
+| ------------ | ------------ | ------------ | ------------ |
+| announcement | news         | contest      | resources    |`,
+    handler: async (ctx) => {
+        const fragList = {
+            announcement: {
+                session: '通知公告',
+                frag: '窗口121',
+            },
+            news: {
+                session: '新闻动态',
+                frag: '窗口123',
+            },
+            contest: {
+                session: '学科竞赛',
+                frag: '窗口124',
+            },
+            resources: {
+                session: '常用资源',
+                frag: '窗口125',
+            },
+        };
+        const type = ctx.req.param('type') ?? 'announcement';
+        const baseUrl = 'http://www.cxcy.ecnu.edu.cn/';
+
+        const response = await got(baseUrl);
+        const $ = load(response.data);
+
+        const filteredEls = $('div.limit_style1 > table > tbody > tr > td')
+            .filter((_, el) => $(el).closest('div').attr('frag') === fragList[type].frag)
```

Comment:
Use the frag value to pinpoint the parent `div` element instead of comparing the parent element for each child element.